controllers/users/users.js

```diff
@@ -118,3 +118,41 @@ exports.deleteUser = (req, res) => {
     .then(() => res.json({ msg: messages.USER_DELETED }))
     .catch(err => res.status(500).send(messages.SERVER_ERROR));
 };
+
+exports.updateUserData = async (req, res) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array({ onlyFirstError: true }) });
+  }
+
+  const user = await User.findById(req.user.id);
+  if (user){
+    let { firstName, lastName, email, password } = req.body;
+
+    user.firstName = firstName || user.firstName;
+    user.lastName = lastName || user.lastName;
+
+    if (email){
+      email = email.toLowerCase();
```

Comment:
we can move: 
 `email = email.toLowerCase();`
to: 
`let { firstName, lastName, email = email.toLowerCase(), password } = req.body;`
and connect 2 condition like : `if (email && email !== user.email)`

Reply:
I was thinking about solution like that but we can't do that because email not always will be specified. and `let { firstName, lastName, email = email.toLowerCase(), password } = req.body;` will throw error. Eventually we cen do this
`let { firstName, lastName, email = "", password } = req.body;`
 `email = email.toLowerCase();`
 `if (email && email !== user.email){`
`  // searching for user with that email and if not found update user email`
 `}`

Reply:
@kamilmoskal email = email.toLowerCase()
if email wont be specific, we got undefined.toLowerCase(), but we cannot make methods on undefined type and we will receive error

Reply:
y i see youre right my bad

Reply:
if(lastName) user.lastName = lastName

maybe it will be better instead of "OR" signature in every fields. What do you think about that solution?

Reply:
@mras2303 yes, that cleaner and more consistent with other checks, thanks